src/modules/clock.cpp

```diff
@@ -1,4 +1,101 @@
 #include "modules/clock.hpp"
+#include <sstream>
+#ifdef HAVE_LANGINFO_1STDAY
+#include <langinfo.h>
+#endif
+
+using zoned_time = date::zoned_time<std::chrono::system_clock::duration>;
+
+struct waybar_time {
+  std::locale locale;
+  zoned_time ztime;
+};
+
+namespace {
+
+#ifdef HAVE_LANGINFO_1STDAY
+// Computations done similarly to Linux cal utility.
+date::weekday first_day_of_week() {
```

Comment:
Something like this would allow correctly rendering calendar with non-default locale. 
```cpp
date::weekday first_day_of_week(const std::locale& locale) {
#ifdef HAVE_LANGINFO_1STDAY
  using locale_ptr_t = std::unique_ptr<std::remove_pointer<locale_t>::type, decltype(&freelocale)>;
  locale_ptr_t loc{newlocale(LC_ALL, locale.name().c_str(), nullptr), &freelocale};
  if (loc) {
    // Computations done similarly to Linux cal utility.
    const int i = (std::intptr_t)nl_langinfo_l(_NL_TIME_WEEK_1STDAY, loc.get());
    auto      ymd = date::year(i / 10000) / (i / 100 % 100) / (i % 100);
    auto      wd = date::weekday(ymd);
    uint8_t   j = *nl_langinfo_l(_NL_TIME_FIRST_WEEKDAY, loc.get());
    return wd + date::days(j - 1);
  }
#endif
  return date::Sunday;
}
```

Reply:
Thanks, missed this, done.

Reply:
Hm, will this work if a user doesn't set a locale in Waybar's clock configuration but sets an environment variable LANG=... instead?  From testing it looks that it works fine but I don't understand how.

Reply:
https://github.com/Alexays/Waybar/blob/83b12fc8a7053d455f37326826120a6f64cf5da6/src/modules/clock.cpp#L12-L16
At least in libstdc++ and libc++ implementations std::locale is built on top of locale_t. And for [newlocale(3)](http://man7.org/linux/man-pages/man3/newlocale.3.html) "" means taking value from environment. Resulting std::locale object will have correct locale name.